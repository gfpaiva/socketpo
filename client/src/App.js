import React, { Component } from 'react';

import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';

import logo from './logo.svg';
import './App.css';

class App extends Component {

  componentWillReceiveProps(nextProps) {
    console.log('ðŸ’£ðŸ’£ðŸ’£ðŸ’£', nextProps);
  }

  createGame = e => {
    e.preventDefault();

    this.props.createGame({
      variables: {
        name: `Match random: ${Math.floor(Math.random() * 100)}`
      }
    })
    .then(res => {
      console.log('âŒ›âŒ›âŒ›âŒ› MUTATION RES', res);
      this.props.getGames.refetch();
    })
  }

  render() {
    const { Games } = this.props.getGames;

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <button onClick={this.createGame}>Create a random named game</button>
        {Games && Games.length > 0 && (
          <div>
            <p>Here are the games folks:</p>
            {Games.map(game => (
              <div key={game.id}>
                <p>
                  <strong>GameID</strong>: {game.id} <br/>
                  <strong>GameHash</strong>: {game.hash} <br/>
                  <strong>GameName</strong>: {game.name} <br/>
                </p>
              </div>
            ))}
          </div>
        )}
      </div>
    );
  }
}

const getGames = gql`
  query getGames {
    Games {
      id
      name
      hash
    }
  }
`;

const createGame = gql`
  mutation createGame($name: String!) {
    createGame(name: $name) {
      id
      hash
      name
    }
  }
`;

const testSub = gql`
  subscription test($id: ID!) {
    justTesting(id: $id) {
      id
      name
      hash
    }
  }
`;

export default compose(
    graphql(getGames, {name: 'getGames'}),
    graphql(createGame, {name: 'createGame'}),
    graphql(testSub, {
      name: 'testSub',
      withRef:true,
      options: {
        variables: { id: 1 }
      }
    })
  )(App);
