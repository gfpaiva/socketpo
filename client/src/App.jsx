import React, { useState, useEffect } from 'react';

import { Route, Switch } from 'react-router-dom';

import NotFound from './Pages/NotFound/NotFound';
import Home from './Pages/Home/Home';
import Create from './Pages/Create/Create';
import Single from './Pages/Single/Single';

import { SoundOn, SoundOff } from './Components/Icons/Icons';


const App = () => {

	const audio = document.querySelector('#audio-initial');
	const [ isMusicMuted, updateMusicMuted ] = useState(false);
	const musicHandler = e => {
		e.preventDefault();
		updateMusicMuted(!isMusicMuted);
	};
	useEffect(() => {
		isMusicMuted ? audio.pause() : audio.play();
	}, [isMusicMuted, audio]);

	useEffect(() => {
		const playPromise = audio.play();

		if (playPromise !== undefined) {
			playPromise.catch(error => {
				updateMusicMuted(true);
			});
		};
	}, [audio]);

	return (
		<div className="App">
			<Switch>
				<Route exact path="/" component={Home}/>
				<Route exact path="/create" component={Create}/>
				<Route exact path="/game/:hash" component={Single}/>

				{/* 404 page */}
				<Route component={NotFound} />
			</Switch>

			<div className="audio-controls">
				<button
					role="img"
					aria-label="Sound"
					className="audio-controls__volume"
					onClick={musicHandler}
				>
					{isMusicMuted ? <SoundOff /> : <SoundOn /> }
				</button>
			</div>

		</div>
	);
};

export default App;
