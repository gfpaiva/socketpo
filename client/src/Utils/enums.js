import React from 'react';
import Avatar from 'avataaars';

import {
	Rock,
	Paper,
	Scissors,
} from '../Components/Icons/Icons';

const statusValues = {
	0: 'Stand By',
	1: 'In progress',
	2: 'Finished'
};

const playValues = {
	0: 'Miss Play',
	1: 'Rock',
	2: 'Paper',
	3: 'Scissors'
};

const playIcons = {
	1: <Rock />,
	2: <Paper />,
	3: <Scissors />
}

const avatarValues = {
	1: <Avatar
		topType='LongHairMiaWallace'
		accessoriesType='Prescription02'
		hairColor='BrownDark'
		facialHairType='Blank'
		clotheType='Hoodie'
		clotheColor='PastelBlue'
		eyeType='Happy'
		eyebrowType='Default'
		mouthType='Smile'
		skinColor='Light'
	/>,
	2: <Avatar
		topType='LongHairBigHair'
		accessoriesType='Blank'
		hairColor='Blonde'
		facialHairType='Blank'
		clotheType='ShirtVNeck'
		clotheColor='PastelYellow'
		eyeType='Wink'
		eyebrowType='RaisedExcitedNatural'
		mouthType='Serious'
		skinColor='DarkBrown'
	/>,
	3: <Avatar
		topType='ShortHairShortFlat'
		accessoriesType='Prescription01'
		hairColor='Blonde'
		facialHairType='Blank'
		clotheType='BlazerSweater'
		eyeType='Default'
		eyebrowType='DefaultNatural'
		mouthType='Serious'
		skinColor='Pale'
	/>,
	4: <Avatar
		topType='ShortHairShortWaved'
		accessoriesType='Blank'
		hairColor='Auburn'
		facialHairType='BeardLight'
		facialHairColor='Red'
		clotheType='BlazerShirt'
		clotheColor='Gray01'
		eyeType='WinkWacky'
		eyebrowType='UnibrowNatural'
		mouthType='Grimace'
		skinColor='DarkBrown'
	/>,
};

export const parseStatus = statusNum => statusValues[statusNum];
export const parsePlay = playNum => playValues[playNum];
export const parsePlayIcons = playNum => playIcons[playNum];
export const parseAvatar = avatarNum => avatarValues[avatarNum];
