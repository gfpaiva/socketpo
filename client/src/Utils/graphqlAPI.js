import gql from 'graphql-tag';

export const totalGameFields = `
name
hash
status
players {
	id
	name
	avatar
	ready
	roundsWin
}
results {
	message
	matchWinner {
		id
		name
		avatar
	}
	rounds {
		isDraw
		finished
		plays {
			player {
				id
				name
			}
			play
		}
		winner {
			id
			name
			avatar
		}
	}
}`;

export const getGame = gql`
	query getGame($hash: String!) {
		GameByHash(hash: $hash) {
			${totalGameFields}
		}
	}
`;

export const play = gql`
	mutation play($hash: String!, $player: PlayerInput!, $play: Int! ) {
		play(hash: $hash, player: $player, play: $play ) {
			hash
			status
		}
	}
`;

export const gameSub = gql`
	subscription gameSub($hash: String!) {
		gameSubscription(hash: $hash) {
			game {
				${totalGameFields}
			}
			message
			hash
			player
		}
	}
`;
