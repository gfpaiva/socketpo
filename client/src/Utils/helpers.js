export const sanitize = str => (str.toLowerCase()
.replace(/[àáâãäå]/ig, 'a')
.replace(/[éèëê]/ig, 'e')
.replace(/[íìïî]/ig, 'i')
.replace(/[óòõöô]/ig, 'o')
.replace(/[uúùüû]/ig, 'u')
.replace(/[ç]/ig, 'c')
.replace(/^[0-9]/ig, '')
.replace(/\s/ig, '-'));
