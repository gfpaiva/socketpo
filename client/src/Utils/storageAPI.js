export const setObject = (key, object) => localStorage.setItem(key, JSON.stringify(object));
export const getObject = key => JSON.parse(localStorage.getItem(key));
