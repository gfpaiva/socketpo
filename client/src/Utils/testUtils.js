export const fakePlayer = (rewrite) => ({
	__typename: 'Player',
	avatar: 3,
	id: 'player01',
	name: 'Player 01',
	ready: false,
	roundsWin: 0,
	...rewrite
});

export const fakeGame = (fullPlayers = true, matchWinner = null, rounds = [], rewrite) => {

	const players = fullPlayers ? [fakePlayer(), fakePlayer({ id: 'player02', name: 'Player 02' })] : [fakePlayer()]

	return {
		__typename: 'Game',
		id: 'game01',
		hash: 'game01',
		name: 'Game 01',
		players,
		results: {
			__typename: "Result",
			matchWinner,
			message: null,
			rounds
		},
		status: 0,
		...rewrite
	};
};

export const fakeRound = (rewrite) => ({
	__typename: 'Round',
	isDraw: false,
	finished: true,
	plays: [
		fakePlay({ play: 1 }),
		fakePlay({ player: fakePlayer({ id: 'player02', name: 'Player 02' }), play: 3 })
	],
	winner: fakePlayer(),
	...rewrite
});

export const fakePlay = (rewrite) => ({
	__typename: 'Play',
	player: fakePlayer(),
	play: 0,
	...rewrite
});
