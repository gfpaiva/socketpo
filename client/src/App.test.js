import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { mountWrap } from './Utils/testUtils'
import App from './App';

// Mock play and pause on audio tags
Object.defineProperty(document, 'querySelector', {
	value: jest.fn().mockReturnValue({ play: jest.fn(), pause: jest.fn() })
});

let wrapper, AppInstance;

beforeEach(() => {
	wrapper = mountWrap(
		<Router>
			<App />
		</Router>
	);

	AppInstance = wrapper.find('App').instance();
});

describe('<App />', () => {
	it('should mount properly and play the audio', () => {
		// Mount and compare snapshot and match if play have been called
		expect(wrapper.find('App')).toMatchSnapshot();
		expect(AppInstance.audio.play).toHaveBeenCalled();
	});

	it('should toggle music state on button click', () => {
		// Initial muted is false
		expect(AppInstance.state.isMusicMuted).toBe(false);

		// Click on CTA button, match if state has beend updated and pause function called
		wrapper.find('.audio-controls__volume').simulate('click');
		expect(AppInstance.state.isMusicMuted).toBe(true);
		expect(AppInstance.audio.pause).toHaveBeenCalled();

		// Click on CTA button, match if state has beend updated and play function called
		wrapper.find('.audio-controls__volume').simulate('click');
		expect(AppInstance.state.isMusicMuted).toBe(false);
		expect(AppInstance.audio.play).toHaveBeenCalled();
	})
});
