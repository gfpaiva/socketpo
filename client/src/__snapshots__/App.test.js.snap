// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should mount properly and play the audio 1`] = `
<App>
  <div
    className="App"
  >
    <RouteWrapper
      atActive={
        Object {
          "opacity": 1,
          "translateX": 0,
        }
      }
      atEnter={
        Object {
          "opacity": 0,
          "translateX": 10,
        }
      }
      atLeave={
        Object {
          "opacity": 0,
          "translateX": -10,
        }
      }
      className="switch-wrapper"
      mapStyles={[Function]}
    >
      <Route>
        <AnimatedSwitch
          atActive={
            Object {
              "opacity": 1,
              "translateX": 0,
            }
          }
          atEnter={
            Object {
              "opacity": 0,
              "translateX": 10,
            }
          }
          atLeave={
            Object {
              "opacity": 0,
              "translateX": -10,
            }
          }
          className="switch-wrapper"
          location={Object {}}
          mapStyles={[Function]}
        >
          <RouteTransition
            atActive={
              Object {
                "opacity": 1,
                "translateX": 0,
              }
            }
            atEnter={
              Object {
                "opacity": 0,
                "translateX": 10,
              }
            }
            atLeave={
              Object {
                "opacity": 0,
                "translateX": -10,
              }
            }
            className="switch-wrapper"
            mapStyles={[Function]}
            runOnMount={false}
            wrapperComponent="div"
          >
            <TransitionMotion
              defaultStyles={null}
              didLeave={[Function]}
              styles={
                Array [
                  Object {
                    "data": <Switch
                      location={Object {}}
                    >
                      <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                      />
                      <Route
                        component={[Function]}
                        exact={true}
                        path="/create"
                      />
                      <Route
                        component={[Function]}
                        exact={true}
                        path="/game/:hash"
                      />
                      <Route
                        component={[Function]}
                      />
                    </Switch>,
                    "key": "",
                    "style": Object {
                      "opacity": Object {
                        "damping": 26,
                        "precision": 0.01,
                        "stiffness": 170,
                        "val": 1,
                      },
                      "translateX": Object {
                        "damping": 26,
                        "precision": 0.01,
                        "stiffness": 170,
                        "val": 0,
                      },
                    },
                  },
                ]
              }
              willEnter={[Function]}
              willLeave={[Function]}
            >
              <div
                className="switch-wrapper"
              >
                <div
                  key=""
                  style={
                    Object {
                      "opacity": 1,
                      "transform": "translateX(0%)",
                    }
                  }
                >
                  <Switch
                    key=""
                    location={Object {}}
                  >
                    <Route
                      component={[Function]}
                      computedMatch={Object {}}
                      location={Object {}}
                    >
                      <NotFound
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={Object {}}
                        match={Object {}}
                      >
                        <h1>
                          Page not foud 
                          <span
                            aria-label="Neutral Face"
                            role="img"
                          >
                            üòê
                          </span>
                        </h1>
                      </NotFound>
                    </Route>
                  </Switch>
                </div>
              </div>
            </TransitionMotion>
          </RouteTransition>
        </AnimatedSwitch>
      </Route>
    </RouteWrapper>
    <div
      className="audio-controls"
    >
      <button
        aria-label="Sound"
        className="audio-controls__volume"
        onClick={[Function]}
        role="img"
      >
        <SoundOn>
          <svg
            version="1.1"
            viewBox="0 0 91 113.75"
            x="0px"
            y="0px"
          >
            <g
              fill="none"
              fillRule="evenodd"
              stroke="none"
              strokeWidth="1"
            >
              <path
                d="M22.6826,0.45508 L22.6826,7.95508 L30.1826,7.95508 L37.6826,7.95508 L45.1826,7.95508 L52.6826,7.95508 L60.1826,7.95508 L67.6826,7.95508 L67.6826,0.45508 L60.1826,0.45508 L52.6826,0.45508 L45.1826,0.45508 L37.6826,0.45508 L30.1826,0.45508 L22.6826,0.45508 L22.6826,0.45508 Z M67.6826,7.95508 L67.6826,15.45508 L75.1826,15.45508 L75.1826,7.95508 L67.6826,7.95508 L67.6826,7.95508 Z M75.1826,15.45508 L75.1826,22.95508 L82.6826,22.95508 L82.6826,15.45508 L75.1826,15.45508 L75.1826,15.45508 Z M82.6826,22.95508 L82.6826,30.45508 L82.6826,37.95508 L82.6826,45.45508 L82.6826,52.95508 L82.6826,60.45508 L82.6826,67.95508 L90.1826,67.95508 L90.1826,60.45508 L90.1826,52.95508 L90.1826,45.45508 L90.1826,37.95508 L90.1826,30.45508 L90.1826,22.95508 L82.6826,22.95508 L82.6826,22.95508 Z M82.6826,67.95508 L75.1826,67.95508 L75.1826,75.45508 L82.6826,75.45508 L82.6826,67.95508 L82.6826,67.95508 Z M75.1826,75.45508 L67.6826,75.45508 L67.6826,82.95508 L75.1826,82.95508 L75.1826,75.45508 L75.1826,75.45508 Z M67.6826,82.95508 L60.1826,82.95508 L52.6826,82.95508 L45.1826,82.95508 L37.6826,82.95508 L30.1826,82.95508 L22.6826,82.95508 L22.6826,90.45508 L30.1826,90.45508 L37.6826,90.45508 L45.1826,90.45508 L52.6826,90.45508 L60.1826,90.45508 L67.6826,90.45508 L67.6826,82.95508 L67.6826,82.95508 Z M22.6826,82.95508 L22.6826,75.45508 L15.1826,75.45508 L15.1826,82.95508 L22.6826,82.95508 L22.6826,82.95508 Z M15.1826,75.45508 L15.1826,67.95508 L7.6826,67.95508 L7.6826,75.45508 L15.1826,75.45508 L15.1826,75.45508 Z M7.6826,67.95508 L7.6826,60.45508 L7.6826,52.95508 L7.6826,45.45508 L7.6826,37.95508 L7.6826,30.45508 L7.6826,22.95508 L0.1826,22.95508 L0.1826,30.45508 L0.1826,37.95508 L0.1826,45.45508 L0.1826,52.95508 L0.1826,60.45508 L0.1826,67.95508 L7.6826,67.95508 L7.6826,67.95508 Z M7.6826,22.95508 L15.1826,22.95508 L15.1826,15.45508 L7.6826,15.45508 L7.6826,22.95508 L7.6826,22.95508 Z M15.1826,15.45508 L22.6826,15.45508 L22.6826,7.95508 L15.1826,7.95508 L15.1826,15.45508 L15.1826,15.45508 Z M52.6826,15.45508 L52.6826,22.95508 L60.1826,22.95508 L60.1826,15.45508 L52.6826,15.45508 L52.6826,15.45508 Z M60.1826,22.95508 L60.1826,30.45508 L67.6826,30.45508 L67.6826,22.95508 L60.1826,22.95508 L60.1826,22.95508 Z M67.6826,30.45508 L67.6826,37.95508 L67.6826,45.45508 L67.6826,52.95508 L67.6826,60.45508 L75.1826,60.45508 L75.1826,52.95508 L75.1826,45.45508 L75.1826,37.95508 L75.1826,30.45508 L67.6826,30.45508 L67.6826,30.45508 Z M67.6826,60.45508 L60.1826,60.45508 L60.1826,67.95508 L67.6826,67.95508 L67.6826,60.45508 L67.6826,60.45508 Z M60.1826,67.95508 L52.6826,67.95508 L52.6826,75.45508 L60.1826,75.45508 L60.1826,67.95508 L60.1826,67.95508 Z M30.1826,22.95508 L30.1826,30.45508 L22.6826,30.45508 L22.6826,37.95508 L15.1826,37.95508 L15.1826,45.45508 L15.1826,52.95508 L22.6826,52.95508 L22.6826,60.45508 L30.1826,60.45508 L30.1826,67.95508 L37.6826,67.95508 L37.6826,60.45508 L37.6826,52.95508 L37.6826,45.45508 L37.6826,37.95508 L37.6826,30.45508 L37.6826,22.95508 L30.1826,22.95508 L30.1826,22.95508 Z M45.1826,30.45508 L45.1826,37.95508 L52.6826,37.95508 L52.6826,30.45508 L45.1826,30.45508 L45.1826,30.45508 Z M52.6826,37.95508 L52.6826,45.45508 L52.6826,52.95508 L60.1826,52.95508 L60.1826,45.45508 L60.1826,37.95508 L52.6826,37.95508 L52.6826,37.95508 Z M52.6826,52.95508 L45.1826,52.95508 L45.1826,60.45508 L52.6826,60.45508 L52.6826,52.95508 L52.6826,52.95508 Z"
                fill="#000000"
              />
            </g>
          </svg>
        </SoundOn>
      </button>
    </div>
  </div>
</App>
`;
